Android 8 and fix problems with other CLLocationManagers from other plugins

t2wu opened the pull request:
Updated config.xml so it can work on Android 8 (Or at least Galaxy S9). And also when there are more than one CLLocationManager used by different plugins, only handles events coming in from our own.

dopry commented:
@t2wu, can you rebase this PR? We fixed up some of the Android 8 issues in another PR.  It would be nice to get you CLLocationManger in a state that is easy to review.

dopry commented:
@t2wu, hey I rebased your changes without the android stuff, but there are still a lot of white space changes. It would also be nice to comment this code. 

Do you think you could rewrite with a short circuit technique so we don't need to re-indent all the code like:

```
// only handle events from our own instance of CLLocationManager
if (manager != self.locationManager) return;
```

petermetz committed:
fix(ios): LocationManager no longer handles other instances messages

This is to avoid conflicts with other plugins which the developer may have installed for their project.

petermetz commented:
Thank you @t2wu and @dopry 
FYI for both: I did a force-push whirlwind while getting used to commit lint, but it's fixed now.

dopry reviewed (commented):
Shouldn't it be 'if (manager != self.locationManager)' !

petermetz commented:
> Shouldn't it be 'if (manager != self.locationManager)' !

Yes it should. :-) Thanks for paying attention, I was very tired last night.

petermetz commented:
@dopry 
FYI: did some more force pushes, watch out.

dopry commented:
No worries force pushes are a part of my day to day.
